autopilot/contractor.go

```diff
@@ -51,85 +53,31 @@ func newContractor(ap *Autopilot) *contractor {
 	}
 }
 
-func (c *contractor) applyConsensusState(cfg Config, state bus.ConsensusState) {
+func (c *contractor) ApplyConsensusState(cfg Config, state bus.ConsensusState) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
 	c.blockHeight = state.BlockHeight
 	if c.currentPeriod == 0 {
 		c.currentPeriod = state.BlockHeight
-	} else if c.blockHeight > c.currentPeriod+cfg.Contracts.Period {
+	} else if c.blockHeight >= c.currentPeriod+cfg.Contracts.Period {
 		c.currentPeriod += cfg.Contracts.Period
 	}
 }
 
-func (c *contractor) contractSpending(contract worker.Contract) (bus.ContractSpending, error) {
-	// TODO: fetch contract hierarchy
-	var contracts []bus.Contract
-
-	// build a map
-	cmap := make(map[types.FileContractID]bus.Contract)
-	for _, contract := range contracts {
-		cmap[contract.ID] = contract
-	}
-
-	// fetch contract chain
-	curr, exists := cmap[contract.ID]
-	if !exists {
-		return bus.ContractSpending{}, fmt.Errorf("contract with id '%v' not found", contract.ID)
-	}
-
-	// no history
-	if curr.RenewedFrom == (types.FileContractID{}) {
-		return curr.Spending, nil
-	}
-
-	// compute total spending
-	total := curr.Spending
-	for exists {
-		if curr, exists = cmap[curr.RenewedFrom]; exists {
-			total = total.Add(curr.Spending)
-		}
-	}
-	return total, nil
-}
-
-func (c *contractor) currentPeriodSpending(contracts []worker.Contract) (types.Currency, error) {
-	totalCosts := make(map[types.FileContractID]types.Currency)
-	for _, c := range contracts {
-		totalCosts[c.ID] = c.TotalCost
-	}
-
-	// filter contracts in the current period
-	var filtered []worker.Contract
-	for _, rev := range contracts {
-		if rev.EndHeight() <= c.currentPeriod {
-			filtered = append(filtered, rev)
-		}
-	}
-
-	// calculate the money spent
-	var spent types.Currency
-	for _, rev := range filtered {
-		remaining := rev.RenterFunds()
-		totalCost := totalCosts[rev.ID]
-		if remaining.Cmp(totalCost) <= 0 {
-			spent = spent.Add(totalCost.Sub(remaining))
-		} else {
-			c.logger.DPanicw("sanity check failed", "remaining", remaining, "totalcost", totalCost)
-		}
-	}
-
-	return spent, nil
+func (c *contractor) CurrentPeriod() uint64 {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	return c.currentPeriod
 }
 
-func (c *contractor) isStopped() bool {
-	select {
-	case <-c.ap.stopChan:
-		return true
-	default:
-		return false
-	}
-}
+func (c *contractor) PerformContractMaintenance(cfg Config) error {
+	// Fetch the consensus dependent fields. We want them to be consistent
```

Comment:
`nit` I think they should/could be passed in here, then it implies what you say in the comment and it does not need a comment